src/components/quantity-counter/QuantitiyCounter.jsx

```diff
@@ -0,0 +1,42 @@
+import AddRoundedIcon from '@mui/icons-material/AddRounded';
+import RemoveRoundedIcon from '@mui/icons-material/RemoveRounded';
+import styles from './QuantitiyCounter.module.css';
+import { useEffect, useState, useRef } from 'react';
+
+export const QuantityCounter = ({ quantityValue = 0 }) => {
+    const [value, setValue] = useState(quantityValue);
+    const decreseButton = useRef();
+    const increaseHandler = () => {
+        setValue((value) => value + 1);
+    }
+
+    const decreaseHandler = () => {
+        setValue((value) => value - 1);
+    }
+
+    useEffect(() => {
+        if (!value) {
+            decreseButton.current.style.pointerEvents = "none";
+            decreseButton.current.style.opacity = "0.2";
+        } else {
+            decreseButton.current.style.pointerEvents = "auto";
+            decreseButton.current.style.opacity = "1";
+        }
```

Comment:
again don't handle the dom save those values in a variable and use it in the element's style prop and you won't need to use ref